### PR TITLE
Add pages add native code and types

### PR DESCRIPTION
**Pull request by vallemar**
This PR addresses a better structure for adding native code

- Java examples have been improved since the existing ones do not work.
- How to enable Kotlin has also been fixed and better explained.
- Now there is also a page for iOS, which for now only has swift code, the importance of @objc has also been briefly explained. I tried objective-c but I don't understand how it works, I have left the section commented, if anyone wants to help here they are welcome.
- On the 'adding native code' page you can now see at a glance that NS supports Java, Kotlin, Objective-C, Swift, SwiftUI and Compose

The code example that currently exists is this
[Java/Kotlin](https://docs.nativescript.org/project-structure/app-resources#adding-native-code-to-an-application). Neither java nor kotlin works. The examples in this PR are tested, copied and pasted.

[Swift/Objective-C](https://docs.nativescript.org/project-structure/app-resources#adding-objectivec-swift-code-to-an-application). This only mentions how to make an extension with Objective-C. As I said in this PR, Objective-C has been eliminated because I can't make it work and I think we should not put examples that don't work (as I said, I have left the section commented, if anyone wants to help, go ahead).

### CODE REVIEW

File: content/guide/native-code/android.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+:::
+
+## Adding Kotlin code
+
+### Kotlin Setting
+
+#### Enable kotlin
+
+If using Kotlin source files, must be enabled.
+
```
Comment (rigor789) [inline content/guide/native-code/android.md:70]:
```suggestion
When using Kotlin, it must be enabled first.
```

File: content/guide/native-code/android.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+Set `useKotlin` in `gradle.properties`, the full path of this file is `App_Resources/Android/gradle.properties`, if it does not exist yet you must create it
```
Comment (rigor789) [inline content/guide/native-code/android.md]:
```suggestion
Set `useKotlin=true` in `App_Resources/Android/gradle.properties` (create the file if it doesn't exist).
```

**Review by rigor789 — COMMENTED** [review]
I see a few more potential improvements, I'll pull this branch and finish them (you can ignore the two suggestions here, I'll push those up as well).

File: content/guide/native-code/android.md (hunk 2 above)
Comment (vallemar) [reply content/guide/native-code/android.md]: @rigor789 Even if `useKotlin=true` is added to the paragraph, I would keep the `properties` block, personally I think it is much more readable to see blocks than to see code in the middle of a paragraph. But if it keeps the block it seems fine to me

Comment (rigor789) [reply content/guide/native-code/android.md]: Yes, both are fine - just simplified wording in the sentence for easier readability.

**Comment by vallemar** [comment]
@rigor789 good review, sorry, my english writing is not my strong point 😢

**Comment by rigor789** [comment]
Your English is perfect, this a very nice change/addition to the docs!

**Pull request merged**